source/common/regular-expressions.js

```diff
@@ -14,6 +14,190 @@
+   */
+  'getCodeBlockRE': function () {
+    return RegExp(
+      /^\s{0,3}(`{3,}|~{3,})/.source
+    )
+  },
+
+  /**
+   * Returns a regular expression that matches footnotes for project export.
+   *
+   * @param   {boolean}  multiline  Whether to match multiline
+   *
+   * @return  {RegExp}              The compiled Regular Expression
+   */
+  'getFnExportRE': function (multiline = false) {
+    let flag = (multiline) ? 'm' : ''
+    return RegExp(
+      /\[\^([\w]+?)\]/.source,
+      // Necessary flags + optional multiline flag
+      'g' + flag)
+  },
+
+  /**
+   * Returns a regular expression that matches MarkDown footnote references.
+   *
+   * @return  {RegExp}              The compiled Regular Expression
+   */
+  'getFnRE': function () {
+    return RegExp(
+      /(\[\^([\da-zA-Z_-]+)\][^:]|\[\^([\da-zA-Z_-]+)\]$)/.source,
+      'g')
+  },
+
+  /**
+   * Returns a regular expression that matches MarkDown footnote references.
+   *
+   * @return  {RegExp}              The compiled Regular Expression
+   */
+  'getFnReferenceRE': function () {
+    return RegExp(
+      /^\[\^.+\]:\s/.source
+    )
+  },
+
+  /**
+   * Returns a regular expression that matches MarkDown footnote references.
+   *
+   * @param   {boolean}  multiline  Whether to match multiline
+   *
+   * @return  {RegExp}              The compiled Regular Expression
+   */
+  'getFnRefRE': function (multiline = false) {
+    let flag = (multiline) ? 'm' : ''
+    return RegExp(
+      /^\[\^([\da-zA-Z_-]+)\]: (.+)/.source,
+      'g' + flag)
+  },
+
+  /**
+   * Returns a regular expression that matches MarkDown footnote references.
+   *
+   * @return  {RegExp}              The compiled Regular Expression
+   */
+  'getFootnoteRefRE': function () {
+    return RegExp(
+      /\[\^[^\]]+\]/.source,
+      'i')
+  },
+
+  /**
+   * Returns a regular expression that matches MarkDown Headings.
+   *
+   * @return  {RegExp}              The compiled Regular Expression
+   */
+  'getHeadRE': function () {
+    return RegExp(
+      /^(#{1,6}) (.*)/.source,
+      'g')
+  },
+
+  /**
+   * Returns a regular expression that matches MarkDown Headings.
+   *
+   * @return  {RegExp}              The compiled Regular Expression
+   */
+  'getHeadingRE': function () {
+    return RegExp(
+      /(#+)\s+/.source
+    )
+  },
+
+  /**
+   * Returns a regular expression that matches MarkDown highlighting (???).
+   *
+   * @return  {RegExp}              The compiled Regular Expression
+   */
+  'getHighlightRE': function () {
+    return RegExp(
+      /::.+?::|==.+?==/.source
+    )
+  },
+
+  /**
+   * Returns a regular expression that matches file IDs as in the settings
+   *
+   * @param   {boolean}  multiline  Whether to match multiline
+   *
+   * @return  {RegExp}              The compiled Regular Expression
+   */
+  'getIDRE': function (multiline = false) {
+    let flag = (multiline) ? 'm' : ''
+    let idRegExpString = global.config.get('zkn.idRE')
```

Comment:
Did you make sure this function doesn't get called before the configuration provider is loaded? (Just asking because I'm always afraid of such race conditions :D)

Reply:
Pretty sure that one was already in this file, but I'll take a look.

Reply:
This code is unchanged.